### PR TITLE
Add utility to check for alphanumeric strings

### PR DESCRIPTION
**Pull request by frank113**
Added the ability to check for alphanumeric strings. 

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing], including the relevant style guides.
-   [x] Read and understand the [Code of Conduct][code-of-conduct].
-   [x] Read and understood the [licensing terms][license].
-   [x] Searched for existing issues and pull requests **before** submitting this pull request.
-   [x] Filed an issue (or an issue already existed) **prior to** submitting this pull request.
-   [x] Rebased onto latest `develop`.
-   [x] Submitted against `develop` branch.

## Description

> What is the purpose of this pull request?

This pull request:

-   Creates functionality to check for alphanumeric strings.

## Related Issues

> Does this pull request have any related issues?

No.

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

[code-of-conduct]: https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md

[license]: https://github.com/stdlib-js/stdlib/blob/develop/LICENSE

### CODE REVIEW

File: lib/node_modules/@stdlib/assert/is-alphanumeric/README.md [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+-->
+
+# isAlphaNumeric
+
+> Test whether a string contains only numeric digits.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/assert/is-alphanumeric/README.md]: Description needs to be updated:

```
Test whether a string contains only alphanumeric characters.
```

File: lib/node_modules/@stdlib/assert/is-alphanumeric/README.md [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+</section>
+
+<!-- /.usage -->
+
+<section class="notes">
+
+## Notes
+
+-   For non-string values, the function returns `false`.
+-   For the purposes of this function, **alphanumeric** is defined to be a collection of letters and numbers.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/assert/is-alphanumeric/README.md]: Can we rephrase to something like `**alphanumeric** is defined as a combination of the letters of the English alphabet and the numbers 0 to 9`?

File: lib/node_modules/@stdlib/assert/is-alphanumeric/docs/repl.txt [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+        Input string
+
+    Returns
+    -------
+    bool: boolean
+        Boolean indicating whether a string contains only alphanumeric characters.
+
+    Examples
+    --------
+    > var bool = {{alias}}( ‘abc0123456789' )
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/assert/is-alphanumeric/docs/repl.txt]: This will not run because of `‘` instead of `'` being used. Should change code to `var bool = {{alias}}( 'abc0123456789' )`

File: lib/node_modules/@stdlib/assert/is-alphanumeric/lib/index.js [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+* @module @stdlib/assert/is-alphanumeric
+*
+* @example
+* var isAlphaNumeric = require( '@stdlib/assert/is-alphanumeric' );
+*
+* var out = isAlphaNumeric( 'abc123a' );
+* // returns true
+*
+* out = isAlphaNumeric( '0xffffff' );
+* // returns trues
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/assert/is-alphanumeric/lib/index.js]: Typo: `s/trues/true`.

File: lib/node_modules/@stdlib/assert/is-alphanumeric/package.json [hunk 5]
```diff
@@ -0,0 +1,81 @@
+{
+  "name": "@stdlib/assert/is-alphanumeric",
+  "version": "0.0.0",
+  "description": "Test whether a string contains only numeric digits.",
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/assert/is-alphanumeric/package.json]: Description needs to be updated:

```
Test whether a string contains only alphanumeric characters.
```

File: lib/node_modules/@stdlib/assert/is-alphanumeric/test/test.js [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+
+// TESTS //
+
+tape( 'main export is a function', function test( t ) {
+	t.ok( true, __filename );
+	t.strictEqual( typeof isAlphaNumeric, 'function', 'main export is a function' );
+	t.end();
+});
+
+tape( 'the function returns `true` if provided a string containing all numeric digits', function test( t ) {
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/assert/is-alphanumeric/test/test.js]: Let's remove this test.

File: lib/node_modules/@stdlib/assert/is-alphanumeric/test/test.js [hunk 7]
```diff
@@ -0,0 +1,125 @@
...
+		'Z'
+	];
+
+	for ( i = 0; i < values.length; i++ ) {
+		t.strictEqual( isAlphaNumeric( values[i] ), true, 'returns true when provided ' + values[i] );
+	}
+	t.end();
+});
+
+tape( 'the function returns `false` if provided a string not containing exclusively alphanumeric characters', function test( t ) {
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/assert/is-alphanumeric/test/test.js]: Test description does not match what is tested, as the description makes it sound like only strings are passed in. Maybe rephrase to `if not provided a string containing exclusively alphanumeric characters`.

**Review by Planeshifter — CHANGES_REQUESTED** [review]
Thanks, Frank!

Flagged some minor issues that need to be resolved before merging in.

File: lib/node_modules/@stdlib/assert/is-alphanumeric/README.md (hunk 1 above)
Comment (frank113) [reply lib/node_modules/@stdlib/assert/is-alphanumeric/README.md]: Resolved!

File: lib/node_modules/@stdlib/assert/is-alphanumeric/README.md (hunk 2 above)
Comment (frank113) [reply lib/node_modules/@stdlib/assert/is-alphanumeric/README.md]: Resolved!

File: lib/node_modules/@stdlib/assert/is-alphanumeric/docs/repl.txt (hunk 3 above)
Comment (frank113) [reply lib/node_modules/@stdlib/assert/is-alphanumeric/docs/repl.txt]: Resolved!

File: lib/node_modules/@stdlib/assert/is-alphanumeric/lib/index.js (hunk 4 above)
Comment (frank113) [reply lib/node_modules/@stdlib/assert/is-alphanumeric/lib/index.js]: Resolved!

File: lib/node_modules/@stdlib/assert/is-alphanumeric/package.json (hunk 5 above)
Comment (frank113) [reply lib/node_modules/@stdlib/assert/is-alphanumeric/package.json]: Resolved!

File: lib/node_modules/@stdlib/assert/is-alphanumeric/test/test.js (hunk 6 above)
Comment (frank113) [reply lib/node_modules/@stdlib/assert/is-alphanumeric/test/test.js]: It was a carryover from copying and pasting. Resolved! The value for that test was merged into the larger "true" tests.

File: lib/node_modules/@stdlib/assert/is-alphanumeric/test/test.js (hunk 7 above)
Comment (frank113) [reply lib/node_modules/@stdlib/assert/is-alphanumeric/test/test.js]: Resolved!

**Review by Planeshifter — APPROVED** [review]
Great, thanks! Merging in after CircleCI has finished.
